bool move_right
bool move_left
bool grasp
bool release
bool assemble_part1
bool assemble_part2
bool assemble_part3